### PR TITLE
[Data][Train] Fix remaining issues on DatasetConfig->DataConfig migration

### PR DESCRIPTION
**Pull request by raulchen**
## Why are these changes needed?

- Change all examples to use `DataConfig`.
- Update function signature of all Trainer classes.
- Add a link in deprecation warning. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-air/doc_code/computer_vision.py [hunk 1]
```diff
@@ -187,7 +187,7 @@ def train_torch_model(dataset, preprocessor, per_epoch_preprocessor):
 
     from ray import train
     from ray.air import session
-    from ray.air.config import DatasetConfig, ScalingConfig
+    from ray.air.config import  ScalingConfig
```
Comment (ericl) [inline doc/source/ray-air/doc_code/computer_vision.py:190]:
```suggestion
    from ray.air.config import ScalingConfig
```

File: doc/source/ray-core/_examples/datasets_train/datasets_train.py [hunk 2]
```diff
@@ -601,6 +602,8 @@ def to_torch_dataset(torch_batch_iterator):
     DROPOUT_EVERY = 5
     DROPOUT_PROB = 0.2
 
+    train_dataset = train_dataset.random_shuffle()
+    test_dataset = test_dataset.random_shuffle()
```
Comment (ericl) [inline doc/source/ray-core/_examples/datasets_train/datasets_train.py:606]: Add a comment that these will be applied on each epoch?

**Review by ericl — APPROVED** [review]

File: doc/source/ray-air/examples/torch_detection.ipynb [hunk 3]
```diff
@@ -823,19 +843,13 @@
                 "    },\n",
                 "    scaling_config=ScalingConfig(num_workers=4, use_gpu=True),\n",
                 "    datasets={\"train\": train_dataset},\n",
-                "    dataset_config={\n",
-                "        # Don't augment test images. Only apply `per_epoch_preprocessor` to the train\n",
-                "        # set.\n",
-                "        \"train\": DatasetConfig(\n",
-                "            per_epoch_preprocessor=per_epoch_preprocessor\n",
-                "        ),\n",
-                "    },\n",
```
Comment (raulchen) [inline doc/source/ray-air/examples/torch_detection.ipynb:832]: @ericl the original comment says “Don't augment test images.“, but there is only a train dataset passed in. Is this an error before?

Comment (ericl) [reply doc/source/ray-air/examples/torch_detection.ipynb:832]: Yeah that seems out of date, we should probably remove the comment.

**Review by ericl — APPROVED** [review]

**Review by c21 — APPROVED** [review]

**Review by pcmoritz — APPROVED** [review]
Great, thanks a ton for doing this :)

**Pull request merged**